kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/PaginatedScanFilesIteratorImpl.java

```diff
@@ -18,14 +18,26 @@
 import io.delta.kernel.PaginatedScanFilesIterator;
 import io.delta.kernel.data.FilteredColumnarBatch;
 import io.delta.kernel.data.Row;
+import io.delta.kernel.internal.util.Utils;
 import io.delta.kernel.utils.CloseableIterator;
 import java.io.IOException;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /** Implementation of {@link PaginatedScanFilesIterator} */
 public class PaginatedScanFilesIteratorImpl implements PaginatedScanFilesIterator {
 
+  private static final Logger logger = LoggerFactory.getLogger(PaginatedScanFilesIteratorImpl.class);
   private final CloseableIterator<FilteredColumnarBatch> originalIterator;
-  private final long pageSize; // max num of files to return in this page
+  private final long pageSize;
+
+  private long numScanFilesReturned = 0;
+  private String currentLogFilePath = null;
+  private long currentRowIdxInFile = 0;
+  private Optional<FilteredColumnarBatch> nextBatch = Optional.empty();
```

Comment:
so we have `currentLogFilePath` and `currentRowIdxInFile` (**"current"**) but then we have `nextBatch` (**"next"**) -- would `currentBatch` be a better name?